Sources/SystemMetrics/SystemMetrics.swift

```diff
@@ -41,8 +68,8 @@ extension MetricsSystem {
     ///     - config: Used to configure `SystemMetrics`.
     public static func bootstrapSystemMetrics(_ config: SystemMetrics.Configuration) {
         self.withWriterLock {
-            precondition(self.systemMetricsProvider == nil, "System metrics already bootstrapped.")
-            self.systemMetricsProvider = SystemMetricsProvider(config: config)
+            precondition(self.systemMetricsProviderContainer.isUninitialized, "System metrics already bootstrapped.")
+            self.systemMetricsProviderContainer.bootstrap(SystemMetricsProvider(config: config))
```

Comment:
Isn't the container using the same lock as we've already acquired here? I.e. we shouldn't acquire the lock here, it should all be done (along with this check) within the container.

Reply:
You're absolutely right, thanks for catching this deadlock-to-be

Reply:
Updated.